### DISCUSSION TITLE
Namespace middleware client shouldn't reconnect on error

### DISCUSSION THREAD

**Discussion by greenbea** [socketio/socket.io#4150] [Q&A]
I have this basic middleware that throws an error when a client's authentication fails. The problem is the clients keep trying to reconnect after getting the `connect_error`. How can I send an error to the client to stop trying to reconnect?

I know I can handle it on the client-side not to reconnect when getting the `connect_error` event, but I want the server-side to handle this, similar to `socket.disconnect()` that causes the client not to try to reconnect.
```
io.use( async (socket, next) => {
    const token = socket.handshake.auth.token
    try {
        let verified = await verifySaveToken(socket, token);
        if (!verified) {
            throw new Error("Token Verification Failed");
        }
        next();
    }
    catch (e) {
        next(new Error(e));
    }
})
```

**Comment by darrachequesne** [comment]
This was not the correct behavior, it should now be fixed: https://github.com/socketio/socket.io-client/commit/d54d12ce634193d14b71894496ed57d35d922378

Thanks for the heads-up :+1:
